usbtool.c

```diff
@@ -27,68 +27,87 @@
 
 enum { INFO, WARNING, ERROR };
 
-void logmsg(int s, char *msg, ...) {
+int isopt(int argc, char **argv, const char *opt){
+
+    for(int i = 1; i < argc; ++i){
+
+        if(strcmp(argv[i], opt) == 0)
+	    return 1;	//opt option has been inputted.
+    }
+
+    return 0;
+}
+
+void logmsg(int state, char *msg, ...) {
     va_list ap;
     time_t t;
     struct tm *l;
     struct utsname u;
     char *err[] = { "", " WARNING:", " ERROR:" };
-
+    
     time(&t);
-    l=localtime(&t);
+    l = localtime(&t);
     uname(&u);
-    if(s>2) 
-        s=2;
-    if(s<0) 
-        s=0;
+    
+    if(state > 2) 
+        state = 2;
+    if(state < 0) 
+        state = 0;
     fflush(stdout);
+    
     printf("%04d/%02d/%02d %02d:%02d:%02d %s:%s ",
-        l->tm_year+1900, l->tm_mon+1, l->tm_mday,
+        l->tm_year + 1900, l->tm_mon + 1, l->tm_mday,
         l->tm_hour, l->tm_min, l->tm_sec,
-        u.nodename, err[s]
+        u.nodename, err[state]
     );
     va_start(ap, msg);
     vprintf(msg, ap);
     va_end(ap);
-    if(s)
+    if(state != INFO)
         printf(" [%s (%d)]", strerror(errno), errno);
     putchar('\n');
     fflush(stdout);
-    if(s==ERROR)
+    if(state == ERROR)
         exit(1);
 }
 
 int main(int argc, char **argv) {
+
     struct udev *udev;
     struct udev_enumerate *enu;
     struct udev_monitor *mon;
     struct udev_device *dev;
     struct udev_list_entry *lst;
-    struct udev_list_entry *entr=NULL;
+    struct udev_list_entry *entr = NULL;
     const char *path, *usbpath, *vendor, *serial, *speed, *action;
     int fd,ret;
     fd_set fds;
 
-    udev=udev_new();
+    if(isopt(argc, argv, "--help")){
```

Comment:
what about `-h` I typically use `-h` instead of `--help`

Reply:
Ok I'll add -h too.

Reply:
Ok I'll add -h too.